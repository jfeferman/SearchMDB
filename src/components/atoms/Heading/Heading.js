import React from 'react';
import PropTypes from 'prop-types';
import {
  StyledH1,
  StyledH2,
  StyledH3,
  StyledH4,
} from './styles';

/**
 * Custom level header
 * @property {string} level - '1' for H1, '2' for H2, '3' for H3, and '4' for H4
 */
const Heading = (props) => {
  const {
    level,
    children,
    className,
  } = props;
  let result = '';
  // Defaults to heading level one
  if (!level || level === '1') {
    result = <StyledH1 className={className}>{children}</StyledH1>;
  } else if (level === '2') {
    result = <StyledH2 className={className}>{children}</StyledH2>;
  } else if (level === '3') {
    result = <StyledH3 className={className}>{children}</StyledH3>;
  } else {
    result = <StyledH4 className={className}>{children}</StyledH4>;
  }
  return result;
};

Heading.propTypes = {
  level: PropTypes.oneOf(['1', '2', '3', '4']),
  children: PropTypes.node,
  className: PropTypes.string,
};

Heading.defaultProps = {
  level: '1',
};

export default Heading;
