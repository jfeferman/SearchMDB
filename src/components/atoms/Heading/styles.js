import styled from 'styled-components';
import { fonts } from '../../styleUtils/globalStyleVariables';

export const StyledH1 = styled.h1`
  ${fonts.H1}
`;

export const StyledH2 = styled.h2`
  ${fonts.H2}
`;

export const StyledH3 = styled.h3`
  ${fonts.H3}
`;

export const StyledH4 = styled.h4`
  ${fonts.H4}
`;
