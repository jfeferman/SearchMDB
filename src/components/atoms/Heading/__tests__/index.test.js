import React from 'react';
import 'jest-styled-components';
import { create } from 'react-test-renderer';
import Heading from '..';
import {
  StyledH1,
  StyledH2,
  StyledH3,
  StyledH4,
} from '../styles';

describe('components/atoms/Heading', () => {
  let renderer;
  let instance;
  let level = '1';
  const text = 'Heading text';

  beforeEach(() => {
    renderer = create(
      <Heading level={level}>
        {text}
      </Heading>,
    );
    instance = renderer.root;
  });

  it('matches the snapshot of the component', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('renders a level 1 heading', () => {
    expect(instance.findAll(el => el.type === StyledH1)).toHaveLength(1);
  });

  it('updates to level 2 and renders the heading', () => {
    level = '2';
    renderer.update(
      <Heading level={level}>
        {text}
      </Heading>,
    );
    expect(instance.findAll(el => el.type === StyledH2)).toHaveLength(1);
  });

  it('updates to level 3 and renders the heading', () => {
    level = '3';
    renderer.update(
      <Heading level={level}>
        {text}
      </Heading>,
    );
    expect(instance.findAll(el => el.type === StyledH3)).toHaveLength(1);
  });

  it('updates to level 4 and renders the heading', () => {
    level = '4';
    renderer.update(
      <Heading level={level}>
        {text}
      </Heading>,
    );
    expect(instance.findAll(el => el.type === StyledH4)).toHaveLength(1);
  });
});
