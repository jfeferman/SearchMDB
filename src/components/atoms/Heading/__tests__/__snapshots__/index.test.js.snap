// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/Heading matches the snapshot of the component 1`] = `
.c0 {
  font-weight: 300;
  font-style: normal;
  color: #000000;
  font-size: 2.8125rem;
  line-height: 3.875rem;
  -webkit-letter-spacing: 0.076875rem;
  -moz-letter-spacing: 0.076875rem;
  -ms-letter-spacing: 0.076875rem;
  letter-spacing: 0.076875rem;
}

@media (max-width:768px) {
  .c0 {
    font-size: 2.125rem;
    line-height: 3rem;
    -webkit-letter-spacing: 0.058125rem;
    -moz-letter-spacing: 0.058125rem;
    -ms-letter-spacing: 0.058125rem;
    letter-spacing: 0.058125rem;
  }
}

<h1
  className="c0"
>
  Heading text
</h1>
`;
